src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,299 @@
+                {
+                    switch (elem.Name.LocalName)
+                    {
+                        case "schema":
+                            // TODO: this
+                            break;
+                        case "assembly":
+                            ParseAssemblyAlias(aliases, elem);
+                            break;
+                        case "resheader":
+                            break;
+                        case "data":
+                            ParseData(filename, pathsRelativeToBasePath, resources, aliases, elem);
+                            break;
+                        default:
+                            throw new NotImplementedException();
+                    }
+                }
+            }
+
+            return resources;
+        }
+
+        private static void ParseAssemblyAlias(Dictionary<string,string> aliases, XElement elem)
+        {
+            string alias = elem.Attribute("alias").Value;
+            string name = elem.Attribute("name").Value;
+
+            aliases.Add(alias, name);
+        }
+
+        // Consts from https://github.com/dotnet/winforms/blob/16b192389b377c647ab3d280130781ab1a9d3385/src/System.Windows.Forms/src/System/Resources/ResXResourceWriter.cs#L46-L63
+        private const string Beta2CompatSerializedObjectMimeType = "text/microsoft-urt/psuedoml-serialized/base64";
+        private const string CompatBinSerializedObjectMimeType = "text/microsoft-urt/binary-serialized/base64";
+        private const string CompatSoapSerializedObjectMimeType = "text/microsoft-urt/soap-serialized/base64";
+        private const string BinSerializedObjectMimeType = "application/x-microsoft.net.object.binary.base64";
+        private const string SoapSerializedObjectMimeType = "application/x-microsoft.net.object.soap.base64";
+        private const string DefaultSerializedObjectMimeType = BinSerializedObjectMimeType;
+        private const string ByteArraySerializedObjectMimeType = "application/x-microsoft.net.object.bytearray.base64";
+        private const string ResMimeType = "text/microsoft-resx";
+
+        private const string StringTypeName = "System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089";
+
+        private static string GetFullTypeNameFromAlias(string aliasedTypeName, Dictionary<string, string> aliases)
+        {
+            if (aliasedTypeName == null)
+            {
+                return StringTypeName;
+            }
+
+            int indexStart = aliasedTypeName.IndexOf(',');
+            if (aliases.TryGetValue(aliasedTypeName.Substring(indexStart + 2), out string fullAssemblyIdentity))
+            {
+                return aliasedTypeName.Substring(0, indexStart + 2) + fullAssemblyIdentity;
+            }
+
+            // Allow "System.String" bare or with no registered alias
+            // TODO: is this overzealous? _Hopefully_ no one is putting a System.String type in their
```

Comment:
Did you have a case where it had an alias and the alias didn't exist?  I'm OK with an equals check here.  If the entry says `System.String` and there is no alias it is reasonable for that to mean the real `System.String`, but if someone says `System.StringComparison` or `System.String, myUtf8String` it's probably not ok to treat it as string.

Reply:
I thought I'd encountered one in MSBuild's tests, but evidently not. Switching to exact-match.